lib/models/device.coffee

```diff
@@ -27,6 +28,37 @@ configModel = require('./config')
 applicationModel = require('./application')
 auth = require('../auth')
 
+MIN_SUPERVISOR_APPS_API = '1.7.1'
+
+###*
+# @summary Check supervisor version compatibility using semver
+# @name isCompatible
+# @private
+# @function
+# @memberof resin.models.device
+#
+# @param {String} version - version under check
+# @param {String} minVersion - minimum accepted version
+# @fulfil {Boolean} - is compatible
+# @reject {Error} Will reject if the given version is < than the given minimum version
+# @returns {Promise}
+#
+# @example
+# resin.models.device.isCompatible(version, MIN_VERSION).then(function(isCompatible) {
+# 	console.log(isCompatible);
+# });
+#
+# @example
+# resin.models.device.isCompatible(version, MIN_VERSION, function(error, isCompatible) {
+# 	if (error) throw error;
+# 	console.log(isCompatible);
+# });
+###
+exports.isCompatible = Promise.method (version, minVersion) ->
```

Comment:
I think we should not export this. In the case of the example, you can just say `isCompatible()`. Also, the fact that the function signature contains `is` implies that the function will return either `true` or `false`, without any side effect, like an error.

What about `ensureSupervisorCompability()`? Clarifying "Supervisor" also make the type of compatibility that we want to check more obvious.